src/cpp/fastdds/xtypes/type_representation/TypeObjectUtils.cpp

```diff
@@ -2050,6 +2050,175 @@ void TypeObjectUtils::plain_collection_type_identifier_header_consistency(
     }
 }
 
+EquivalenceKind TypeObjectUtils::get_map_component_equiv_kind_for_consistency(
+        const TypeIdentifier& identifier,
+        bool is_key)
+{
+    if (is_direct_hash_type_identifier(identifier))
+    {
+        TypeObject type_object;
+        if (eprosima::fastdds::dds::RETCODE_OK ==
+                type_object_registry_observer().get_type_object(identifier, type_object))
+        {
+            return type_object._d();
```

Comment:
This is not really checking if the key type is correct. If `is_key` is true, the received TypeObject MUST be of type ALIAS and such alias MUST be for one of the supported key types.

Reply:
changed in [a989bd3](https://github.com/eProsima/Fast-DDS/pull/4496/commits/a989bd32df198abf53da0faff2c924f05a240c47)